policy/handler.go

```diff
@@ -5,173 +5,97 @@ package policy
 
-	// doneCh is used to signal the handler to stop.
-	doneCh chan struct{}
+	statusGetter targetpkg.TargetStatusGetter
 
-	// reloadCh is used to communicate to the MonitorPolicy routine that it
-	// should perform a reload.
-	reloadCh chan struct{}
+	// cooldownCh is used to notify the handler that it should enter a cooldown
+	// period.
+	cooldownCh <-chan time.Duration
+	errChn     chan<- error
+}
+
+//	StartNewHandler starts the handler for the given policy
+//
+// This function blocks until the context provided is canceled.
+type HandlerConfig struct {
+	CooldownChan chan time.Duration
+	UpdatesChan  chan *sdk.ScalingPolicy
+	ErrChan      chan<- error
+	Policy       *sdk.ScalingPolicy
+	Log          hclog.Logger
+	TargetGetter targetpkg.TargetStatusGetter
 }
 
-// NewHandler returns a new handler for a policy.
-func NewHandler(ID PolicyID, log hclog.Logger, pm *manager.PluginManager, ps Source) *Handler {
-	return &Handler{
-		policyID:      ID,
-		log:           log.Named("policy_handler").With("policy_id", ID),
-		pluginManager: pm,
-		policySource:  ps,
+func RunNewHandler(ctx context.Context, evalCh chan<- *sdk.ScalingEvaluation,
+	config HandlerConfig) {
+
+	h := &Handler{
+		log: config.Log.Named("policy_handler").With("policy_id", config.Policy.ID),
 		mutators: []Mutator{
 			NomadAPMMutator{},
 		},
-		ch:         make(chan sdk.ScalingPolicy),
-		errCh:      make(chan error),
-		doneCh:     make(chan struct{}),
-		cooldownCh: make(chan time.Duration, 1),
-		reloadCh:   make(chan struct{}),
+		cooldownCh:   config.CooldownChan,
+		statusGetter: config.TargetGetter,
+		updatesCh:    config.UpdatesChan,
+		policy:       config.Policy,
+		errChn:       config.ErrChan,
 	}
-}
-
-// Run starts the handler and periodically sends the policy for evaluation.
-//
-// This function blocks until the context provided is canceled or the handler
-// is stopped with the Stop() method.
-func (h *Handler) Run(ctx context.Context, evalCh chan<- *sdk.ScalingEvaluation) {
-	h.log.Trace("starting policy handler")
-
-	defer h.Stop()
-
-	// Mark the handler as running.
-	h.runningLock.Lock()
-	h.running = true
-	h.runningLock.Unlock()
-
-	// Store a local copy of the policy so we can compare it for changes.
-	var currentPolicy *sdk.ScalingPolicy
 
-	// Start with a long ticker until we receive the right interval.
-	// TODO(luiz): make this a config param
-	policyReadTimeout := 3 * time.Minute
-	h.ticker = time.NewTicker(policyReadTimeout)
-
-	// Create separate context so we can stop the monitoring Go routine if
-	// doneCh is closed, but ctx is still valid.
-	monitorCtx, cancel := context.WithCancel(ctx)
-	defer cancel()
-
-	// Start monitoring the policy for changes.
-	req := MonitorPolicyReq{ID: h.policyID, ErrCh: h.errCh, ReloadCh: h.reloadCh, ResultCh: h.ch}
-	go h.policySource.MonitorPolicy(monitorCtx, req)
+	h.ticker = time.NewTicker(h.policy.EvaluationInterval)
+	defer h.ticker.Stop()
 
+	h.log.Error("starting policy handler")
```

Comment:
Probably shouldn't be an error :grin: 

```suggestion
	h.log.Trace("starting policy handler")
```